### PR TITLE
Filter ListValidators by Active in RPC

### LINKED ISSUE #4029: GetValidators RPC call returns all the Validators in State
Currently calling `GetValidators` returns all the validators stored in our registry instead of just the active validators. This issue is to track the resolution of this bug

### PR DESCRIPTION
**Pull request by rauljordan**
Resolves #4029 and depends on #4059 

---

# Description

**Write why you are making the changes in this pull request**

This PR uses the `Active bool` field in `ListValidatorsRequest` to filter out only active validators in the request for validators via RPC.

**NOTE: Do not merge until #4059 is checked in**

### CODE REVIEW

File: beacon-chain/rpc/beacon/validators.go [hunk 1]
```diff
@@ -187,18 +187,18 @@ func (bs *Server) ListValidators(
...
+		for idx, val := range vals {
 			// The first time we see a validator with an activation epoch > the requested epoch,
 			// we know this validator is from the future relative to what the request wants.
 			if val.ActivationEpoch > requestedEpoch {
 				stopIdx = idx
 				break
 			}
 		}
-		validators = validators[:stopIdx]
+		vals = vals[:stopIdx]
```
Comment (shayzluf) [inline beacon-chain/rpc/beacon/validators.go]: why do we need this line?

Comment (shayzluf) [reply beacon-chain/rpc/beacon/validators.go]: got it nm

File: beacon-chain/rpc/beacon/validators.go [hunk 2]
```diff
@@ -209,8 +209,20 @@ func (bs *Server) ListValidators(
 		)
 	}
 
-	validatorCount := len(validators)
-	// If there are no validators, we simply return a response specifying this.
+	// Filter active validators if the request specifies it.
+	res := vals
+	if req.Active {
+		filteredValidators := make([]*ethpb.Validator, 0)
+		for _, val := range vals {
```
Comment (shayzluf) [inline beacon-chain/rpc/beacon/validators.go]: we can filter in the above loop

Comment (rauljordan) [reply beacon-chain/rpc/beacon/validators.go]: Hey @shayzluf I tried to do it in the above loop but that's only for archival data, so I would need another loop for non-archival requests anyways, creating more code than necessary. This seemed to be the shortest approach.

**Pull request merged**